### DISCUSSION TITLE
[Android][iOS] Geolocator.PositionChanged doesn't trigger

### DISCUSSION THREAD

**Discussion by mcocaroUX** [unoplatform/uno#16079] [Q&A] [answered]
### Current behavior

Event `PositionChanged` never gets triggered when subscribed (tested on emulators and physical devices)

### Expected behavior

`PositionChanged` should trigger when user moves

### How to reproduce it (as minimally and precisely as possible)

[GeoTest.zip](https://github.com/unoplatform/uno/files/14794800/GeoTest.zip)

Geolocator.Position is being subscribed at MainModel constructor. There you can notice how on iOS and Android Locations list never gets updated.

### Works on UWP/WinUI

Yes

### NuGet package version(s)

 <ItemGroup>
   <PackageVersion Include="Microsoft.ApplicationInsights" Version="2.22.0" />
   <PackageVersion Include="Microsoft.Extensions.Logging.ApplicationInsights" Version="2.22.0" />
   <PackageVersion Include="Microsoft.Extensions.Logging.Console" Version="8.0.0" />
   <PackageVersion Include="Microsoft.Windows.SDK.BuildTools" Version="10.0.22621.2428" />
   <PackageVersion Include="Microsoft.WindowsAppSDK" Version="1.4.231219000" />
   <PackageVersion Include="Microsoft.Maui.Essentials" Version="8.0.3" />
  
   <PackageVersion Include="Serilog" Version="3.1.1" />
   <PackageVersion Include="Serilog.Extensions.Hosting" Version="8.0.0" />
   <PackageVersion Include="Serilog.Extensions.Logging" Version="8.0.0" />
   <PackageVersion Include="Serilog.Sinks.ApplicationInsights" Version="4.0.0" />
   <PackageVersion Include="Serilog.Sinks.Console" Version="5.0.1" />
  
   <PackageVersion Include="Uno.Core.Extensions.Logging.Singleton" Version="4.0.1" />
   <PackageVersion Include="Uno.Extensions.Authentication" Version="4.0.0" />
   <PackageVersion Include="Uno.Extensions.Authentication.WinUI" Version="4.0.0" />
   <PackageVersion Include="Uno.Extensions.Configuration" Version="4.0.0" />
   <PackageVersion Include="Uno.Extensions.Hosting" Version="4.0.0" />
   <PackageVersion Include="Uno.Extensions.Hosting.WinUI" Version="4.0.0" />
   <PackageVersion Include="Uno.Extensions.Http" Version="4.0.0" />
   <PackageVersion Include="Uno.Extensions.Http.WinUI" Version="4.0.0" />
   <PackageVersion Include="Uno.Extensions.Http.Refit" Version="4.0.0" />
   <PackageVersion Include="Uno.Extensions.Localization" Version="4.0.0" />
   <PackageVersion Include="Uno.Extensions.Localization.WinUI" Version="4.0.0" />
   <PackageVersion Include="Uno.Extensions.Logging.OSLog" Version="1.7.0" />
   <PackageVersion Include="Uno.Extensions.Logging.WinUI" Version="4.0.0" />
   <PackageVersion Include="Uno.Extensions.Navigation" Version="4.0.0" />
   <PackageVersion Include="Uno.Extensions.Navigation.WinUI" Version="4.0.0" />
   <PackageVersion Include="Uno.Extensions.Navigation.Toolkit.WinUI" Version="4.0.0" />
   <PackageVersion Include="Uno.Extensions.Reactive" Version="4.0.0" />
   <PackageVersion Include="Uno.Extensions.Reactive.WinUI" Version="4.0.0" />
   <PackageVersion Include="Uno.Extensions.Serialization" Version="4.0.0" />
   <PackageVersion Include="Uno.Extensions.Serialization.Http" Version="4.0.0" />
   <PackageVersion Include="Uno.Extensions.Serialization.Refit" Version="4.0.0" />
   <PackageVersion Include="Uno.Extensions.Reactive.Messaging" Version="2.5.11" />
  
   <PackageVersion Include="Uno.Material.WinUI" Version="4.1.0" />
   <PackageVersion Include="Uno.Dsp.Tasks" Version="1.2.8" />
   <PackageVersion Include="Uno.Toolkit.WinUI" Version="5.1.5" />
   <PackageVersion Include="Uno.Toolkit.WinUI.Material" Version="5.1.5" />
   <PackageVersion Include="Uno.Resizetizer" Version="1.2.1" />
   <PackageVersion Include="Uno.UniversalImageLoader" Version="1.9.36" />
   <PackageVersion Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="$(UnoVersion)" />
   <PackageVersion Include="Uno.WinUI" Version="$(UnoVersion)" />
   <PackageVersion Include="Uno.WinUI.Lottie" Version="$(UnoVersion)" />
   <PackageVersion Include="Uno.WinUI.RemoteControl" Version="$(UnoVersion)" />
   <PackageVersion Include="Uno.WinUI.DevServer" Version="$(UnoVersion)" />
   <PackageVersion Include="Uno.WinUI.Svg" Version="$(UnoVersion)" />
   <PackageVersion Include="Uno.WinUI.DevServer" Version="$(UnoVersion)" />
  
   <PackageVersion Include="Mapsui.Uno.WinUI" Version="4.1.2" />
   <PackageVersion Include="CommunityToolkit.Mvvm" Version="8.2.0" />
   <PackageVersion Include="SkiaSharp.Views.Uno" Version="2.88.6" />
   <PackageVersion Include="Xamarin.Google.Android.Material" Version="1.9.0.2" />
   <PackageVersion Include="Xamarin.Android.Google.BillingClient" Version="6.0.1.4" />
   <PackageVersion Include="Xamarin.Android.Google.BillingClient.Ktx" Version="6.0.1.4" />
   <PackageVersion Include="Xamarin.AndroidX.Legacy.Support.V4" Version="1.0.0.18" />
   <PackageVersion Include="Plugin.InAppBilling" Version="7.1.0" />
 </ItemGroup>

### Affected platforms

Android, iOS

### IDE

Visual Studio 2022

**Comment by jeromelaban** [comment]
Thanks for the report. GeoLocator is functioning properly, you can find an example as shown in the Uno Gallery here:

https://github.com/unoplatform/Uno.Gallery/blob/7559ae7f013aa3d2c8964ad1ccd21c5ce93255eb/Uno.Gallery/Uno.Gallery.Shared/Views/SamplePages/GeolocatorSamplePage.xaml.cs#L44

Note that you have to request for access first.

**Reply by mcocaroUX** [reply]
@jeromelaban I noted that on the sample is not requesting for permission, I updated here: 
[GeoTest (2).zip](https://github.com/unoplatform/uno/files/14838701/GeoTest.2.zip)

This was detected on our app, that we have the propper permissios on AndroidManifest and Info.plist, and I'm requesting for user permission before interacting with the Geolocator. We are successfully getting the current location by using `GetGeopositionAsync` but when subscribing to the `PositionChanged` event, I dont get any position updates. I've checked the sample and seems to be aligned

**Reply by jeromelaban** [reply]
The original sample gets the updated positions, so there's likely something else missing in your version of the app. @ajpinedam can you try it out?

**Reply by mcocaroUX** [reply]
@jeromelaban I retested the sample and it only get the position updates on windows. It doesn't seem to work on mobile platforms

**Reply by ajpinedam** [reply]
I found a workaround that seems to work for both mobile platforms (iOS and Android). 

Before subscribing to the `PositionChanged` event, call the `GetGeopositionAsync()` method on the Geolocator instance.

```csharp
if (accessStatus == GeolocationAccessStatus.Allowed)
{
    _ = await _geolocator.GetGeopositionAsync();

    _geolocator.PositionChanged += Locator_PositionChanged;
}
```

I am still looking for a final solution.

cc: @mcocaroUX

**Reply by mcocaroUX** [reply]
I will try that, thanks @ajpinedam

**Reply by mcocaroUX** [reply]
@ajpinedam Seems that this workaround only works for Android and not on iOS

**Reply by ajpinedam** [reply, marked as answer]
@mcocaroUX really!? Sorry, didn't validate on iOS. 

The good thing is that a final fix was merged earlier this week and this should be available in the latest dev-built. This fix cover both platforms (iOS and Android)

PR: https://github.com/unoplatform/uno/pull/16209
